### PR TITLE
(Probably not for submission!) Example of how to configure an application for Data Protection

### PR DESCRIPTION
**Pull request by jskeet**
- This example assumes it can use default credentials
- Documentation would need to mention the permissions required for the service account for wherever the code is running
- I want to make the options handling here much simpler

### CODE REVIEW

File: GuessWord/Startup.cs [hunk 1]
```diff
@@ -46,6 +48,15 @@ public void ConfigureServices(IServiceCollection services)
                 options.MinimumSameSitePolicy = SameSiteMode.None;
             });
 
+            // Configure Data Protection to store keys in Google Cloud Storage, encrypted using Google Cloud KMS.
+            // These keys are then used to provide anti-forgery tokens which are consistent across servers.
+            var storageOptions = Configuration.GetSection("GoogleCloudStorageOptions").Get<GoogleCloudStorageOptions>();
+            var kmsOptions = Configuration.GetSection("GoogleCloudKmsOptions").Get<GoogleCloudKmsOptions>();
```
Comment (SurferJeffAtGoogle) [inline GuessWord/Startup.cs]: This looks really pretty!

Comment (jskeet) [reply GuessWord/Startup.cs]: This part isn't *too* bad - particularly if we put `GoogleCloudStorageOptions` as a class in `Google.Cloud.AspNetCore.DataProtection.Storage`, and add an overload that allows it to be passed in directly.
But I'm still concerned about the lack of being able to specify a `StorageClient` or credentials. I'm hoping to ask a Stack Overflow question today.

**Comment by jskeet** [comment]
Note that I don't have access to merge this change myself. We probably want to change the config files not to refer to my project and bucket if we're going to merge it :)

**Comment by jskeet** [comment]
I've tweaked the settings in a third commit, so the PR should be fine to merge now if you wish to.

**Pull request merged**